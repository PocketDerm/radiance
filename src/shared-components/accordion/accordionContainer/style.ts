import styled from '@emotion/styled';
import { BOX_SHADOWS, BREAKPOINTS, COLORS } from 'src/constants';

const Container = styled.div<{ borderRadius?: string }>`
  border-radius: ${({ borderRadius }) => borderRadius || 'inherit'};
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export default { Container };
