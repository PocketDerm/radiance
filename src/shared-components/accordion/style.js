import styled from '@emotion/styled';
import { css } from '@emotion/core';

import {
  ANIMATION,
  BOX_SHADOWS,
  BREAKPOINTS,
  COLORS,
  SPACER,
} from '../../constants';

const border = `1px solid ${COLORS.border}`;

export const Container = styled.div`
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = isOpen => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div`
  ${({ noBorder, isOpen }) => !noBorder && getBorderStyle(isOpen)};
  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled &&
    `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `};
`;

export const ArrowWrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${SPACER.medium};
`;

export const TitleWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;
