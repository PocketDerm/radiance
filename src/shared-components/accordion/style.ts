import styled from '@emotion/styled';
import { css } from '@emotion/core';
import {
  ANIMATION,
  BREAKPOINTS,
  BOX_SHADOWS,
  COLORS,
  SPACER,
} from 'src/constants';

const border = `1px solid ${COLORS.border}`;

const getTopBorderRadius = (borderRadius: string) => ({
  borderTopLeftRadius: borderRadius,
  borderTopRightRadius: borderRadius,
});

const getBottomBorderRadius = (borderRadius: string) => ({
  borderBottomLeftRadius: borderRadius,
  borderBottomRightRadius: borderRadius,
});

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div<{ contentHeight: string }>`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = (isOpen: boolean) => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div<{
  noBorder: boolean;
  isOpen: boolean;
  disabled: boolean;
}>`
  ${({ noBorder, isOpen }) => (!noBorder ? getBorderStyle(isOpen) : '')};

  position: relative;

  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled
      ? `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `
      : ''};
`;

export const ArrowWrapper = styled.div<{ rightAlign: boolean }>`
  display: flex;
  align-items: center;

  ${({ rightAlign }) =>
    rightAlign
      ? `padding-left: ${SPACER.medium};`
      : `padding: 0 ${SPACER.medium};`};
`;

export const TitleWrapper = styled.div<{
  borderRadius: string;
  disabled: boolean;
  isOpen: boolean;
}>`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  &:focus {
    outline: none;
    box-shadow: ${BOX_SHADOWS.focusSecondary};
  }

  div:last-of-type & {
    &:focus {
      ${({ borderRadius, isOpen }) => {
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return (
      !isOpen &&
          `
        border-bottom-left-radius: ${borderBottomLeftRadius}; 
        border-bottom-right-radius: ${borderBottomRightRadius};
        `
    );
  }}
    }
  }
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;

/**
 * borderRadius must match borderRadius passed to main <Accordion />
 * component if opting out of default values.
 */
export const Container = styled.div<{ borderRadius?: string }>`
  box-shadow: ${BOX_SHADOWS.clickable};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;

  ${({ borderRadius = '4px' }) => {
    const { borderTopLeftRadius, borderTopRightRadius } = getTopBorderRadius(
      borderRadius,
    );
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return `
    > div:first-of-type {
      border-top-left-radius: ${borderTopLeftRadius};
      border-top-right-radius: ${borderTopRightRadius};

      ${TitleWrapper} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }

      ${AccordionBox} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }
    }
  
    > div:last-of-type {
      border-bottom-left-radius: ${borderBottomLeftRadius};
      border-bottom-right-radius: ${borderBottomRightRadius};
  
      ${AccordionBox} {
        border-bottom-left-radius: ${borderBottomLeftRadius};
        border-bottom-right-radius: ${borderBottomRightRadius};
      }
    }
`;
  }}
`;
