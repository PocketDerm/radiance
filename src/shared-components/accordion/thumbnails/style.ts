import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../../typography';
import { COLORS, SPACER } from '../../../constants';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  margin-left: ${SPACER.large};
`;

const thumbnailBase = css`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 1.5rem;
  width: 1rem;
`;

export const ImageContainer = styled.div`
  ${thumbnailBase} overflow: hidden;
  height: 1.5rem;
  width: 1rem;

  &:not(:last-child) {
    margin-right: 4px;
  }
`;

export const ThumbnailImage = styled.img`
  min-height: 1.5rem;
  min-width: 1rem;
  object-fit: cover;
`;

export const MultiplesContainer = styled.div`
  ${thumbnailBase}
  background-color: ${COLORS.purple15};
`;

export const MultiplesText = styled.div`
  ${TYPOGRAPHY_STYLE.label};
  font-size: 10px;
`;
