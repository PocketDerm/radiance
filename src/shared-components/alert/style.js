import styled from '@emotion/styled';
import { css, keyframes } from '@emotion/core';

import {
  COLORS,
  MEDIA_QUERIES,
  SPACER,
  ANIMATION,
  TYPOGRAPHY_CONSTANTS,
} from '../../constants';

export const AlertsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: fixed;
  top: ${SPACER.medium};
  left: 0;
  right: 0;
  z-index: 99999;

  ${MEDIA_QUERIES.mdUp} {
    left: auto;
    right: ${SPACER.medium};
  }
`;

const fadeInDesktop = keyframes`
  from { opacity: 0; transform: translate3d(24px, 0, 0); }
  to { opacity: 1; transform: translate3d(0, 0, 0); }
`;

const fadeInMobile = keyframes`
  from { opacity: 0; transform: translate3d(0, -24px, 0);  }
  to { opacity: 1; transform: translate3d(0, 0px, 0); }
`;

const defaultAlertStyles = css`
  background-color: ${COLORS.primary};
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
`;

const successAlertStyles = css`
  background-color: ${COLORS.success};
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
`;

const errorAlertStyles = css`
  background-color: ${COLORS.error};
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
`;

export const AlertContainer = styled.div`
  cursor: pointer;
  position: relative;
  margin: 0 auto ${SPACER.small};
  padding: 0;
  width: 327px;
  border-radius: ${SPACER.small};
  opacity: ${props => (props.exiting ? '0' : '1')};
  animation: ${fadeInMobile} ${ANIMATION.defaultTiming} 1;
  transition: ${ANIMATION.defaultTiming};
  transform: ${({ exiting }) =>
    exiting ? 'translate3d(0, -24px, 0)' : 'translate3d(0, 0, 0)'};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return errorAlertStyles;
      case 'error':
        return errorAlertStyles;
      case 'success':
        return successAlertStyles;
      default:
        return defaultAlertStyles;
    }
  }};

  ${MEDIA_QUERIES.mdUp} {
    animation: ${fadeInDesktop} ${ANIMATION.defaultTiming} 1;
    margin: 0 auto ${SPACER.medium};
    transform: ${({ exiting }) =>
      exiting ? 'translate3d(24px, 0, 0)' : 'translate3d(0, 0, 0)'};
  }
`;

export const MainContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  padding: ${SPACER.medium};
`;

export const ContentContainer = styled.div`
  margin: -3px 0 0 ${SPACER.medium};
  max-height: 48px;
`;

export const CtaContent = styled.div`
  padding: ${SPACER.medium};

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'error':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'success':
        return css`
          border-top: 1px solid ${COLORS.successBorder};
        `;
      default:
        return css`
          border-top: 1px solid ${COLORS.infoBorder};
        `;
    }
  }};
`;

export const IconContainer = styled.div`
  ${({ hasAvatar }) =>
    hasAvatar &&
    css`
      min-width: 32px;
    `};

  svg {
    height: ${SPACER.medium};
    width: ${SPACER.medium};
    fill: ${COLORS.white};
  }
`;
