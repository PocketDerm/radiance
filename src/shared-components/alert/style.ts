import styled from '@emotion/styled';
import { css, keyframes, SerializedStyles } from '@emotion/core';

import {
  COLORS,
  MEDIA_QUERIES,
  SPACER,
  ANIMATION,
  TYPOGRAPHY_CONSTANTS,
} from '../../constants';

export const AlertsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: fixed;
  top: ${SPACER.medium};
  left: 0;
  right: 0;
  z-index: 99999;

  ${MEDIA_QUERIES.mdUp} {
    left: auto;
    right: ${SPACER.medium};
  }
`;

const fadeInDesktop = keyframes`
  from { opacity: 0; transform: translate3d(24px, 0, 0); }
  to { opacity: 1; transform: translate3d(0, 0, 0); }
`;

const fadeInMobile = keyframes`
  from { opacity: 0; transform: translate3d(0, -24px, 0);  }
  to { opacity: 1; transform: translate3d(0, 0px, 0); }
`;

const defaultAlertStyles = css`
  background-color: ${COLORS.primary};
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
`;

const successAlertStyles = css`
  background-color: ${COLORS.success};
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
`;

const errorAlertStyles = css`
  background-color: ${COLORS.error};
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
`;

export const AlertContainer = styled.div<{
  exiting: boolean;
  alertType: string;
}>`
  cursor: pointer;
  position: relative;
  margin: 0 auto ${SPACER.small};
  padding: 0;
  width: 327px;
  border-radius: ${SPACER.small};
  opacity: ${({ exiting }): string => (exiting ? '0' : '1')};
  animation: ${fadeInMobile} ${ANIMATION.defaultTiming} 1;
  transition: ${ANIMATION.defaultTiming};
  transform: ${({ exiting }): string =>
    exiting ? 'translate3d(0, -24px, 0)' : 'translate3d(0, 0, 0)'};

  ${({ alertType }): SerializedStyles => {
    switch (alertType) {
      case 'danger':
        return errorAlertStyles;
      case 'error':
        return errorAlertStyles;
      case 'success':
        return successAlertStyles;
      default:
        return defaultAlertStyles;
    }
  }};

  ${MEDIA_QUERIES.mdUp} {
    animation: ${fadeInDesktop} ${ANIMATION.defaultTiming} 1;
    margin: 0 auto ${SPACER.medium};
    transform: ${({ exiting }): string =>
    exiting ? 'translate3d(24px, 0, 0)' : 'translate3d(0, 0, 0)'};
  }
`;

export const MainContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  padding: ${SPACER.medium};
`;

export const ContentContainer = styled.div<{ truncateText: boolean }>`
  margin: -3px 0 0 ${SPACER.medium};
  max-height: ${({ truncateText }): string => (truncateText ? '48px' : 'none')};
`;

export const CtaContent = styled.div`
  padding: ${SPACER.medium};

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  border-top: 1px solid rgba(255, 255, 255, 0.1);
`;

export const IconContainer = styled.div<{ hasAvatar: boolean }>`
  ${({ hasAvatar }): SerializedStyles | false =>
    hasAvatar &&
    css`
      min-width: 32px;
    `};

  svg {
    height: ${SPACER.medium};
    width: ${SPACER.medium};
    fill: ${COLORS.white};
  }
`;
