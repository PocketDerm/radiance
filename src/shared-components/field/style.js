import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../typography';
import { COLORS, BOX_SHADOWS, SPACING, ANIMATION } from '../../constants';

export const HintItem = styled.div`
  ${TYPOGRAPHY_STYLE.caption};
  margin: 0;
  transition: all ${ANIMATION.defaultTiming};
  opacity: 0;
  max-height: 0;
`;

export const FieldContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: ${SPACING.small};
`;

export const Label = styled.label`
  ${TYPOGRAPHY_STYLE.label};

  ${({ disabled }) => disabled && `color:${COLORS.purple30};`};
`;

const inputStyles = css`
  ${TYPOGRAPHY_STYLE.body};
  appearance: none;
  background: ${COLORS.white};
  border: 1px solid ${COLORS.border};
  border-radius: 0;
  color: ${COLORS.primaryTint1};
  padding: 13px ${SPACING.small};
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
  outline: none;
  margin-bottom: ${SPACING.xsmall};

  &:active,
  &:focus {
    outline: none;
    border-color: ${COLORS.primary};
    box-shadow: ${BOX_SHADOWS.focusSecondary};

    ~ ${HintItem} {
      max-height: 23px;
      opacity: 1;
    }
  }

  &::placeholder {
    color: ${COLORS.primaryTint3};
  }

  &[disabled] {
    background-color: ${COLORS.disabled};
    color: ${COLORS.textDisabled};
    cursor: not-allowed;
    border: 1px solid ${COLORS.border};
    box-shadow: none;
  }
`;

export const Input = styled.input`
  ${inputStyles};
  padding-right: ${SPACING.medium};
`;

export const Textarea = styled.textarea`
  ${inputStyles};
  color: ${COLORS.primary};
  display: block;
  height: 100%;
  line-height: ${SPACING.base};
  margin: 0 auto;
  max-width: 35rem;
  padding: ${SPACING.small};
  resize: none;
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
`;

const errorStyles = css`
  svg.error-icon {
    opacity: 1;
  }

  ${Textarea}, ${Input} {
    border-color: ${COLORS.statusRed};

    &:active,
    &:focus {
      border-color: ${COLORS.statusRed};
      box-shadow: none;
    }
  }
`;

export const InputContainer = styled.div`
  position: relative;

  svg.error-icon {
    opacity: 0;
    position: absolute;
    top: 20px;
    right: ${SPACING.small};
    width: 16px;
    height: 16px;
    fill: ${COLORS.statusRed};
    transition: opacity ${ANIMATION.defaultTiming};
  }

  ${({ showErrors }) => showErrors && errorStyles};
`;
