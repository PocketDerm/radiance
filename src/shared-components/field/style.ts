import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../typography';
import {
 COLORS, BOX_SHADOWS, SPACER, ANIMATION, 
} from '../../constants';
import { MessagesTypes } from '../verificationMessages';

export const HintItem = styled.div`
  ${TYPOGRAPHY_STYLE.caption};
  transition: all ${ANIMATION.defaultTiming} ease-in-out;
  opacity: 0;
  max-height: 0;
`;

export const FieldContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 22px;
`;

export const Label = styled.label<{ disabled: boolean }>`
  ${TYPOGRAPHY_STYLE.label};

  ${({ disabled }) => disabled && `color:${COLORS.primaryTint3};`};
`;

const inputStyles = css`
  ${TYPOGRAPHY_STYLE.body};
  appearance: none;
  background: ${COLORS.white};
  border: 1px solid ${COLORS.border};
  border-radius: ${SPACER.xsmall};
  color: ${COLORS.primaryTint1};
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
  outline: none;
  margin-bottom: 2px;

  &:active,
  &:focus {
    outline: none;
    border-color: ${COLORS.primary};
    box-shadow: ${BOX_SHADOWS.focus};

    ~ ${HintItem} {
      max-height: 24px;
      opacity: 1;
      margin: ${SPACER.x2small} 0 0 0;
    }
  }

  &::placeholder {
    color: ${COLORS.primaryTint3};
  }

  &[disabled] {
    background-color: ${COLORS.disabled};
    color: ${COLORS.textDisabled};
    cursor: not-allowed;
    border: 1px solid ${COLORS.border};
    box-shadow: none;
  }
`;

export const Input = styled.input`
  ${inputStyles};
  padding: 13px ${SPACER.xlarge} 13px ${SPACER.medium};
`;

export const Textarea = styled.textarea`
  ${inputStyles};
  color: ${COLORS.primary};
  display: block;
  height: 100%;
  line-height: ${SPACER.large};
  margin: 0 auto;
  max-width: 35rem;
  padding: ${SPACER.medium};
  resize: none;
  transition: border-color ${ANIMATION.defaultTiming};
  width: 100%;
`;

const applyMessagesStyles = (messagesType: MessagesTypes) => css`
  svg.radiance-field-input-icon {
    opacity: 1;
  }

  ${Textarea}, ${Input} {
    border-color: ${messagesType === 'success' ? COLORS.success : COLORS.error};

    &:active,
    &:focus {
      border-color: ${messagesType === 'success'
        ? COLORS.success
        : COLORS.error};
      box-shadow: ${BOX_SHADOWS.focus};
    }
  }
`;

export const InputContainer = styled.div<{
  messagesType: MessagesTypes;
  showMessages: boolean;
}>`
  position: relative;

  svg.radiance-field-input-icon {
    opacity: 0;
    position: absolute;
    top: 20px;
    right: ${SPACER.medium};
    transition: opacity ${ANIMATION.defaultTiming};
  }

  ${({ showMessages, messagesType }) =>
    showMessages && applyMessagesStyles(messagesType)};
`;
