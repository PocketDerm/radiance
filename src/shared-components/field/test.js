import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { ErrorItem } from './style';

import Field from './index';

describe('<Field />', () => {
  describe('UI Snapshot', () => {
    it('renders with label prop', () => {
      const labelText = 'Test Label';
      const component = renderer.create(
        <Field label={labelText}>
          <Field.Input />
        </Field>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders with label and labelFor', () => {
      const labelText = 'Test Label';
      const labelFor = 'for-input-id';
      const component = renderer.create(
        <Field label={labelText} labelFor={labelFor}>
          <Field.Input />
        </Field>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when provided an error object show the message', () => {
    it('renders the error messages', () => {
      const errors = { maxLength: 'Maximum 6 characteres' };
      const wrapper = mount(
        <Field errors={errors}>
          <Field.Input />
        </Field>
      );
      const li = wrapper.find(ErrorItem);

      expect(li).toHaveLength(1);
      expect(li.text().match(errors.maxLength)).toBeTruthy();
    });
  });
});
