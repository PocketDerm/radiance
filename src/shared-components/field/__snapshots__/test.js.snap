// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> UI Snapshot renders with label and labelFor 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1.5rem;
}

.emotion-0 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
}

.emotion-4 {
  position: relative;
}

.emotion-4 svg {
  opacity: 0;
  position: absolute;
  top: 20px;
  right: 1rem;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-2 {
  color: #524D6E;
  font-size: 1rem;
  line-height: 1.75;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #ffffff;
  border: 1px solid #ededf0;
  border-radius: 0.25rem;
  color: #524D6E;
  -webkit-transition: border-color 350ms;
  transition: border-color 350ms;
  width: 100%;
  outline: none;
  padding: 13px 2rem 13px 1rem;
}

.emotion-2:active,
.emotion-2:focus {
  outline: none;
  border-color: #332e54;
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
}

.emotion-2:active ~ .ez9grqt0,
.emotion-2:focus ~ .ez9grqt0 {
  max-height: 23px;
  opacity: 1;
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  margin: 0.125rem 0 0 0;
}

.emotion-2::-webkit-input-placeholder {
  color: #c3c0cd;
}

.emotion-2::-moz-placeholder {
  color: #c3c0cd;
}

.emotion-2:-ms-input-placeholder {
  color: #c3c0cd;
}

.emotion-2::placeholder {
  color: #c3c0cd;
}

.emotion-2[disabled] {
  background-color: #ededf0;
  color: #c3c0cd;
  cursor: not-allowed;
  border: 1px solid #ededf0;
  box-shadow: none;
}

<div
  className="emotion-6 emotion-7"
>
  <label
    className="emotion-0 emotion-1"
    disabled={false}
    htmlFor="for-input-id"
  >
    Test Label
  </label>
  <div
    className="emotion-4 emotion-5"
  >
    <svg
      fill="#BD200F"
    />
    <input
      className="emotion-2 emotion-3"
      disabled={false}
    />
  </div>
</div>
`;

exports[`<Field /> UI Snapshot renders with label prop 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1.5rem;
}

.emotion-0 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
}

.emotion-4 {
  position: relative;
}

.emotion-4 svg {
  opacity: 0;
  position: absolute;
  top: 20px;
  right: 1rem;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-2 {
  color: #524D6E;
  font-size: 1rem;
  line-height: 1.75;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #ffffff;
  border: 1px solid #ededf0;
  border-radius: 0.25rem;
  color: #524D6E;
  -webkit-transition: border-color 350ms;
  transition: border-color 350ms;
  width: 100%;
  outline: none;
  padding: 13px 2rem 13px 1rem;
}

.emotion-2:active,
.emotion-2:focus {
  outline: none;
  border-color: #332e54;
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
}

.emotion-2:active ~ .ez9grqt0,
.emotion-2:focus ~ .ez9grqt0 {
  max-height: 23px;
  opacity: 1;
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  margin: 0.125rem 0 0 0;
}

.emotion-2::-webkit-input-placeholder {
  color: #c3c0cd;
}

.emotion-2::-moz-placeholder {
  color: #c3c0cd;
}

.emotion-2:-ms-input-placeholder {
  color: #c3c0cd;
}

.emotion-2::placeholder {
  color: #c3c0cd;
}

.emotion-2[disabled] {
  background-color: #ededf0;
  color: #c3c0cd;
  cursor: not-allowed;
  border: 1px solid #ededf0;
  box-shadow: none;
}

<div
  className="emotion-6 emotion-7"
>
  <label
    className="emotion-0 emotion-1"
    disabled={false}
    htmlFor="Test Label"
  >
    Test Label
  </label>
  <div
    className="emotion-4 emotion-5"
  >
    <svg
      fill="#BD200F"
    />
    <input
      className="emotion-2 emotion-3"
      disabled={false}
    />
  </div>
</div>
`;
